docs/tutorials/writing_tutorials.md

```diff
@@ -0,0 +1,307 @@
+```
+╼> go run test/utils/tutorials/tester.go --verbose --run-only writing_tutorials.md
+# Testing writing_tutorials.md
+
+
+## stage init with 1 chunks
+
+
+## stage test1 with 1 chunks
+
+ SUCCESS  echo a line starting with dollar gets executed, and its output is shown under in /tmp/3361630480
+ INFO  a line starting with dollar gets executed, and its output is shown under
+
+```
+
+We can see that the `init` chunk we've created at the beginning of the document
+is also taken into account, even though there's no command to execute it that
+chunk.
+
+Let's add a bit of complexity, we will create two chunks, the first one extract
+several variables and the second one consumes one of them.
+
+````
+```{"stage":"test2", "runtime":"bash", "variables":["VARIABLE", "TEST", "EDITOR"]}
+$ VARIABLE=$(sleep .1s && echo "This is some content")
+$ TEST="some value"
+```
+````
+
+````
+```{"stage":"test2", "env":["VARIABLE"], "runtime":"bash"}
+$ if [ "$VARIABLE" == "This is some content" ]; then
+$   echo "same string"
+$ fi
+
+same string
+```
+````
+
+This is getting us:
+
+```
+## stage test2 with 2 chunks
+
+ SUCCESS  ./scriptc0ea8682-5725-40b1-b3af-895b10538147.sh in /tmp/2216367646
+ INFO  Extracted variable: $VARIABLE="This is some content"
+ INFO  Extracted variable: $TEST="some value"
+ INFO  Extracted variable: $EDITOR="nvim"
+ SUCCESS  ./scriptf6c0cf71-e244-4301-a51a-3d471bc44b13.sh in /tmp/1769345675 with env [VARIABLE=This is some content]
+ INFO  out: same string
+```
+
+Let's add some parallelism. A stage will perform changes to a common
+file with two concurrent processes. Then a last stage will collect the changes
+and display them.
+
+````
+```{"stage":"test3", "runtime":"bash", "parallel":true, "rootdir":"$tmpdir.1"}
+$ for i in $(seq 1 10);
+$ do
+$     echo FIRST$i >> output
+$     sleep .1
+$ done
+```
+````
+
+````
+```{"stage":"test3", "runtime":"bash", "parallel":true, "rootdir":"$tmpdir.1"}
+$ for i in $(seq 1 10);
+$ do
+$     sleep .1
+$     echo SECOND$i >> output
+$ done
+```
+````
+
+````
+```{"stage":"test4", "rootdir":"$tmpdir.1"}
+$ cat output
+
+FIRST1
+SECOND1
+FIRST2
+SECOND2
+FIRST3
+FIRST4
+SECOND3
+FIRST5
+SECOND4
+FIRST6
+SECOND5
+FIRST7
+SECOND6
+SECOND7
+FIRST8
+SECOND8
+FIRST9
+FIRST10
+SECOND9
+SECOND10
+```
+````
+
+This gives us the result:
+
+```
+## stage test3 with 2 chunks
+
+ SUCCESS  ./scriptbdfcb2a4-495a-4a1b-9ccc-bf0545d6e58a.sh in /tmp/577314846
+ SUCCESS  ./script0e605abe-f19c-4fe8-ac9d-448d836f2945.sh in /tmp/577314846
+
+## stage test4 with 2 chunks
+
+ SUCCESS  cat output in /tmp/577314846
+ INFO  FIRST1
+       SECOND1
+       FIRST2
+       SECOND2
+       FIRST3
+       FIRST4
+       SECOND3
+       FIRST5
+       SECOND4
+       FIRST6
+       SECOND5
+       FIRST7
+       SECOND6
+       SECOND7
+       FIRST8
+       SECOND8
+       FIRST9
+       FIRST10
+       SECOND9
+       SECOND10
+```
+
+Using HereTags:
+
+````
+```{"stage":"test4", "runtime":"bash", "HereTag":"EOF"}
+$ cat << EOF
```

Comment:
Could the tutorial tester autodetect command lines that ends with `<< EOF`?

Reply:
I see your point but I suppose that that tutorial test could restrict the supported here tags to `<< EOF` simpolifing the usage without limiting too much the feature itself, unless you are aware of use case that could require another here tag.